### DISCUSSION TITLE
Is there a way to remove or customize the datetime string in the version string?

### DISCUSSION THREAD

**Discussion by chshouyu** [tauri-apps/tauri#8191] [Q&A]
For example, the version number is: Version 1.2.2 (20230908.045356), I want to remove `(20230908.045356)` or customize it.
Thanks!

**Comment by xueweiwujxw** [comment]
How did you add it? I saw that the deb control only has the version number and no timestamp.

**Reply by chshouyu** [reply]
It's on macOS

**Reply by xueweiwujxw** [reply]
Oh, thanks.
I foung another solution. I use `genver.sh` to generate version json file and edit the `tauri.config.json` to use the `version` of generated file.

`genver.sh`
```shell
cat <<EOF >.env.version.json
{
  "version": "$(jq -r .version package.json)-$(date +%s)-$(git rev-parse --short=8 HEAD)"
}
EOF
```
`tauri.config.json`
```json
{
  "package": {
    "productName": "xxxxx",
    "version": "../.env.version.json"
  },
}
```

`version` has a hint: 
App version. It is a semver version number or a path to a `package.json` file containing the `version` field. If removed the version number from `Cargo.toml` is used.

I edit the `package.json` to ensure my script has been excuted.
`package.json`
```json
{
  "scripts": {
    "predev": "cross-os dev",
    "prebuild": "cross-os build",
    "dev": "vite",
    "build": "tsc && vite build",
    "preview": "vite preview",
    "tauri": "tauri",
    "pretauri": "cross-os tauri"
  },
  "cross-os": {
    "dev": {
      "darwin": "sh ./scripts/dev.env.sh",
      "linux": "sh ./scripts/dev.env.sh",
      "win32": "pwsh ./scripts/dev.env.ps1"
    },
    "build": {
      "darwin": "sh ./scripts/pro.env.sh",
      "linux": "sh ./scripts/pro.env.sh",
      "win32": "pwsh ./scripts/pro.env.ps1"
    },
    "tauri": {
      "darwin": "sh ./scripts/genver.sh",
      "linux": "sh ./scripts/genver.sh",
      "win32": "pwsh ./scripts/genver.ps1"
    }
  },
  "devDependencies": {
    "cross-os": "^1.5.0",
  }
}
```
Also I added `cross-os` to dev dependencies
